kore/src/Kore/AST/ApplicativeKore.hs

```diff
@@ -0,0 +1,87 @@
+{-|
+Module      : Kore.AST.ApplicativeKore
+Description : TBA.
+Copyright   : (c) Runtime Verification, 2018
+-}
+module Kore.AST.ApplicativeKore
+    ( completeDefinition
+     )
+where
+
+import Control.Comonad
+import Data.Foldable
+import Kore.AST.Common
+import Kore.AST.MetaOrObject
+import Kore.AST.Sentence
+import Kore.AST.Valid
+import Kore.Step.Pattern hiding
+       ( freeVariables )
+
+-- use (CommonStepPattern level) for patternType
+
+completeDefinition :: VerifiedPureDefinition Object -> VerifiedPureDefinition Object
+completeDefinition Definition { definitionAttributes, definitionModules } =
+    Definition
+    { definitionAttributes
+    , definitionModules = map completeModule definitionModules
+    }
+
+completeModule :: VerifiedPureModule Object -> VerifiedPureModule Object
+completeModule Module { moduleName, moduleSentences, moduleAttributes } =
+    Module
+    { moduleName
+    , moduleSentences = concat (map completeSentence moduleSentences)
+    , moduleAttributes}
+
+completeSentence :: VerifiedPureSentence Object -> [VerifiedPureSentence Object]
+completeSentence =
```

Comment:
I would use separate clauses here instead of a `\case`,
```.hs
completeSentence (SentenceAxiomSentence sentenceAxiom) = ...
completeSentence s = [s]
```